### PR TITLE
test: additional unit tests for SPD and DIR metrics

### PR DESCRIPTION
**Pull request by adolfo-ab**
Adds a few unit tests for SPD and DIR metrics, along with a few smalls refactor into existing tests (mostly pre-commit formatting).

## Summary by Sourcery

Add comprehensive unit tests for Disparate Impact Ratio and Statistical Parity Difference metrics using Hypothesis and synthetic data, refactor test data generation and introduce helper functions for outcome manipulation, and update dependencies.

Enhancements:
- Refactor synthetic data setup into a reusable generate_data function and add helper functions for setting group outcomes

Build:
- Add hypothesis and pytest to project dependencies in pyproject.toml

Tests:
- Add scenario-based and property-based unit tests for Disparate Impact Ratio covering reference consistency, positivity, zero-case, equality-case, and empty inputs
- Add scenario-based and property-based unit tests for Statistical Parity Difference covering consistency with reference, valid range, zero equality, sign indication, and empty inputs

### CODE REVIEW

File: tests/metrics/test_fairness.py [hunk 1]
```diff
@@ -12,44 +14,52 @@
...
-    "HasCrCard": np.random.randint(0, 2, n_rows),
-    "IsActiveMember": np.random.randint(0, 2, n_rows),
-    "Exited": np.random.randint(0, 2, n_rows),
-    "Complain": np.random.randint(0, 2, n_rows),
-    "Satisfaction Score": np.random.randint(1, 6, n_rows),
-}
+from hypothesis import given, strategies as st, settings, Verbosity
 
+
+# generate synthetic bank churn data for testing
```
Comment (ruivieira) [inline tests/metrics/test_fairness.py:31]: Perhaps for a future PR we could use something like `sklearn.datasets.make_classification` to simulate data with biased distributions

**Review by ruivieira — APPROVED** [review]

File: tests/metrics/test_fairness.py (hunk 1 above)
Comment (adolfo-ab) [reply tests/metrics/test_fairness.py:31]: I take note of this, will do in an upcoming PR
